parameters:
  # save settings
  global_dir: "/home/yixfeng/Block-SLAM/map/M2DGR/street_01/"
  downsample_resolution: 0.2

  # lidar settings
  odom_child_frame_id: "velodyne"
  lidarTopic: "/velodyne_points"
  have_ring_time_channel: false     # if "XYZI", false; if "XYZIRT", true
  lidarMinRange: 2.0
  lidarMaxRange: 80.0

  # imu settings
  imuTopic: "/handsfree/imu"
  imuAccNoise: 0.01
  imuGyrNoise: 0.01
  gravity: 9.80416
  imuFrequency: 150

  # sensors extrinsics
  LiDAR2Imu_TransV: [ -0.29, 0, -0.15 ]
  LiDAR2Imu_RotV: [ 1, 0, 0,
                    0, 1, 0,
                    0, 0, 1 ]
  
  LiDAR2GT_TransV: [ 0.16, 0, -0.15 ]
  LiDAR2GT_RotV: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1 ]

  # keyframe settings
  trans_detla: 0.5
  rot_detla: 0.2

gtsam:
  # gtsam settings
  odomLinearNoise: 0.08
  odomAngularNoise: 0.05
  active_factors: 120
  preserve_factors: 10

localization:
  # interval settings
  cool_time_duration: 2.00
  key_interval: 0.1

  # localization settings
  ndt_neighbor_search_method: "DIRECT1"
  ndt_resolution: 2.0
  ndt_epsilon: 0.01
  penalty_threshold: 0.6
  penalty_weight: 1.0